vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java

```diff
@@ -151,33 +170,44 @@ public void close() {
     Bson bquery = wrap(query);
     Bson bupdate = wrap(update);
     if (options.isMulti()) {
-      coll.updateMany(bquery, bupdate, mongoUpdateOptions(options), convertCallback(resultHandler, result -> null));
+      coll.updateMany(bquery, bupdate, mongoUpdateOptions(options), toMongoClientUpdateResult(resultHandler));
     } else {
-      coll.updateOne(bquery, bupdate, mongoUpdateOptions(options), convertCallback(resultHandler, result -> null));
+      coll.updateOne(bquery, bupdate, mongoUpdateOptions(options), toMongoClientUpdateResult(resultHandler));
     }
     return this;
   }
 
-  @Override
+  @Deprecated @Override
   public io.vertx.ext.mongo.MongoClient replace(String collection, JsonObject query, JsonObject replace, Handler<AsyncResult<Void>> resultHandler) {
     replaceWithOptions(collection, query, replace, DEFAULT_UPDATE_OPTIONS, resultHandler);
     return this;
   }
 
   @Override
+  public MongoClient replaceDocuments(String collection, JsonObject query, JsonObject replace, Handler<AsyncResult<MongoClientUpdateResult>> resultHandler) {
+    replaceDocumentsWithOptions(collection, query, replace, DEFAULT_UPDATE_OPTIONS, resultHandler);
+    return this;
+  }
+
+  @Deprecated @Override
   public io.vertx.ext.mongo.MongoClient replaceWithOptions(String collection, JsonObject query, JsonObject replace, UpdateOptions options, Handler<AsyncResult<Void>> resultHandler) {
+    return replaceDocumentsWithOptions(collection, query, replace, options, toVoidAsyncResult(resultHandler));
+  }
+
+  @Override
+  public MongoClient replaceDocumentsWithOptions(String collection, JsonObject query, JsonObject replace, UpdateOptions options, Handler<AsyncResult<MongoClientUpdateResult>> resultHandler) {
     requireNonNull(collection, "collection cannot be null");
     requireNonNull(query, "query cannot be null");
     requireNonNull(replace, "update cannot be null");
     requireNonNull(options, "options cannot be null");
     requireNonNull(resultHandler, "resultHandler cannot be null");
 
     boolean id = query.containsKey(ID_FIELD);
-    JsonObject encodedQuery = encodeKeyWhenUseObjectId(query);
+    query = encodeKeyWhenUseObjectId(query);  //TODO: Need to write test for this and delete
```

Comment:
Yes that would be fine, it can go in the backlog.

Cheers,
Martijn

On 23 May 2016 at 11:39, Maziz <notifications@github.com> wrote:

> In
> vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java
> <https://github.com/vert-x3/vertx-mongo-client/pull/78#discussion_r64201552>
> :
>
> >      requireNonNull(collection, "collection cannot be null");
> >      requireNonNull(query, "query cannot be null");
> >      requireNonNull(replace, "update cannot be null");
> >      requireNonNull(options, "options cannot be null");
> >      requireNonNull(resultHandler, "resultHandler cannot be null");
> >
> >      boolean id = query.containsKey(ID_FIELD);
> > -    JsonObject encodedQuery = encodeKeyWhenUseObjectId(query);
> > +    query = encodeKeyWhenUseObjectId(query);  //TODO: Need to write test for this and delete
>
> I think should add this one in another ticket. Enhancement maybe?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/vert-x3/vertx-mongo-client/pull/78/files/fa1765b808fff985421fc4e09102fe11ae618e24#r64201552>
>